### DISCUSSION TITLE
ViCare ZK03840 thermostat

### DISCUSSION THREAD

**Discussion by linbench** [Koenkk/zigbee2mqtt#14101] [Devices]
Hello, I've some radiator thermostats from viessmann (ViCare ZK03840). I am using them together with home assistant. So far they are working, but I can't switch the system mode back to sleep over the UI. It works direct over mqtt message, but this results in an wrong state in home assistant. I tried an update of zigbee2mqtt but that did not change the actual behavior. Only thing I realized is that after the restart of zigbee2mqtt after the update, the sleep button was shown in the UI from zigbee2mqtt device detail view. But it will dissapear afeter some time again. And also home assistant has due to that only the heat system mode available. I realy would like to fix that, could someone point me to the correct direction where to look at.

**Comment by linbench** [comment]
Hello, I've some new info. Yesterday I found some log in Home Assistant that says that 'sleep' is an invalid argument for 'system_mode'.
**2022-09-20 21:01:20.529 ERROR (MainThread) [homeassistant.components.mqtt.climate] Invalid modes mode: sleep**
Then I was looking for an option to change the climate controll to be able to also use the 'sleep' argument. Sadly it looks like that that is not possible with the climate controll. Then I was looking at the converter and for test I changed the 'sleep' to 'off' and in that case home assistant was happy and showed me also the off button at the climate widget. Of course the thermostat then complained about an unknown command over zigbee. At that point I was getting to 'toZigbee' I was looking at the converters for an solution. But there I don't know how to do it. The idea would be to convert the 'off' to an 'sleep' toZigbee and the opposite way back. Can someone help me with that. Or pont me at the correct direction to solve my proble. 

Thank you

**Comment by ArtCore7** [comment]
Just got my Viessmans now i am waiting for Raspbee and when everything is here i will face the same Problems... 😉 If I can find a workaround, or you already got one would be nice to Share.

Just started to search the Code and found this, maybe we need a MR in HA:

`if(mode.values.includes('sleep')) {
                    // 'sleep' is not supported by homeassistent, but is valid according to ZCL
                    // TRV that support sleep (e.g. Viessmann) will have it removed from here,
                    // this allows other expose consumers to still use it, e.g. the frontend.
                    mode.values.splice(mode.values.indexOf('sleep'), 1);
                }`

**Reply by diricxbart** [reply]
Hello @kayweber , @ArtCore7,
I'm also interested in setting up this thermostat with home assistant. Could you give some instructions on how you accomplished this?
Thanks,
Bart

**Comment by linbench** [comment]
Nice that not only me has this issue. Unfortnatly I don't have time right now too look deeper into this. I hope I can start again at this soon.

**Comment by ArtCore7** [comment]
I did not go deeper in. I got some Workarounds with Better Thermostat and some custom YAML Config and it works...

**Reply by linbench** [reply]
Sounds great, could you share that part with the custom YAML Config or even better an little description with your solution. Would be great also for others that try to get this thermostat working.
